### PR TITLE
Add different period for google drive

### PR DESCRIPTION
**Pull request by icecat2012**

### CODE REVIEW

**Comment by icecat2012** [comment]
the parent folder name of the image files should be '1day' or '3day' or '7day' or '14day'. If the parent folder was not named above, the default time will be set as 7day.

File: arrange_schedule.py [hunk 1]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+    pure_result = db.query(sql)
+    try:
+        return_msg['data_type'] = int(pure_result[0][0])
+        return_msg['result'] = 'success'
+    except:
+        return_msg["error"] = "no google_drive_image data type"
+    db.close()
+    return return_msg
+
+def serach_google_drive_folder(service):
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach`

File: arrange_schedule.py [hunk 2]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+    g_sql = "(name='1day' or name='3day' or name='7day' or name='14day')"
+    results = service.files().list(
+        q="mimeType='application/vnd.google-apps.folder' and trashed=false and " + g_sql, 
+        fields="nextPageToken, files(id, name)").execute()
+    items = results.get('files', [])
+    return items
+
+def serach_google_drive_file(service):
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach`

File: arrange_schedule.py [hunk 3]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+        drive_file[num1]['time'] = 86400*7
+        for item in days_limit:
+            if item['id'] in drive_file[num1]['parents']:
+                drive_file[num1]['time'] = 86400 * int(item['name'][:-3])
+                break
+    return drive_file
+
+def check_drive_img_exist(db, data_type, file_name):
+    sql = "SELECT COUNT(*) FROM image_data WHERE img_is_expire=0 and img_is_delete=0 "
+    sql = sql + "and type_id=" + str(data_type) + " and img_file_name='" + file_name + "'"
```
Comment (Billy4195) [inline arrange_schedule.py]: Please modify the string usage to use `format` method.
Refer to [format example](https://docs.python.org/3.5/library/string.html#format-examples)

File: arrange_schedule.py [hunk 4]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+    return pure_result[0][0]
+
+def save_google_drive_file(service, json_obj):
+    try:
+        db = mysql()
+        db.connect()
+        return_msg={}
+        return_msg['result'] = 'fail'
+        for item in json_obj['files']:
+            #print('##{0} ({1})'.format(item['name'], item['id']))
```
Comment (Billy4195) [inline arrange_schedule.py]: The commenst for test should be deleted.

File: arrange_schedule.py [hunk 5]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+            send_obj["file_type"] = json_obj['data_type']
+            send_obj["file_dir"] = 'static/img/' + file_name
+            send_obj["start_date"] = time.strftime("%Y-%m-%d", time.localtime(time.time()))
+            send_obj["end_date"] = time.strftime("%Y-%m-%d", time.localtime(time.time()+item['time']))
+            send_obj["start_time"] = "00:00:00"
+            send_obj["end_time"] = "23:59:59"
+            send_obj["display_time"] = 3
+            send_obj["user_id"] = json_obj['user_id']
+            receive_obj = upload_image_insert_db(send_obj)
+            #pprint(receive_obj)
```
Comment (Billy4195) [inline arrange_schedule.py]: Same as above

File: arrange_schedule.py [hunk 6]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+            #save thumbnail image
+            try:
+                if receive_obj["result"] == "success":
+                    filepath = receive_obj["img_system_dir"]
+                    thumbnail_path = "static/thumbnail/"
+                    thumbnail_path = os.path.join(thumbnail_path,receive_obj["img_thumbnail_name"])
+                    im = Image.open(filepath)
+                    im.thumbnail((100,100))
+                    im.save(thumbnail_path)
+                    #print(thumbnail_path)
```
Comment (Billy4195) [inline arrange_schedule.py]: Save as above

File: arrange_schedule.py [hunk 7]
```diff
@@ -1063,97 +1184,32 @@ def crawler_google_drive_img(json_obj):
...
-                        return_msg = receive_obj
-                        return return_msg
-                except:
-                    db.close()
-                    return_msg["error"] = "save thumbnail image fail"
-                    return return_msg
-                
-        db.close()
+        #serach_google_drive
+        json_obj['files'] = serach_google_drive(service)
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach` on the above.

File: arrange_schedule.py [hunk 8]
```diff
@@ -1063,97 +1184,32 @@ def crawler_google_drive_img(json_obj):
...
-                        db.close()
-                        return_msg = receive_obj
-                        return return_msg
-                except:
-                    db.close()
-                    return_msg["error"] = "save thumbnail image fail"
-                    return return_msg
-                
-        db.close()
+        #serach_google_drive
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach` on the above.

File: arrange_schedule.py [hunk 9]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+        return_msg['result'] = 'success'
+        return return_msg
+    except Exception as e:
+        db.close()
+        return_msg["error"] = str(e)
+        return return_msg
+
+def serach_google_drive(service):
+    # find folder
+    days_limit = serach_google_drive_folder(service)
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach` on the above.

File: arrange_schedule.py [hunk 10]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+                return return_msg
+        db.close()
+        return_msg['result'] = 'success'
+        return return_msg
+    except Exception as e:
+        db.close()
+        return_msg["error"] = str(e)
+        return return_msg
+
+def serach_google_drive(service):
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach` on the above.

File: arrange_schedule.py [hunk 11]
```diff
@@ -1049,7 +1049,128 @@ def check_event_exist_or_insert(event):
...
+    # find folder
+    days_limit = serach_google_drive_folder(service)
+
+    # find file
+    drive_file = serach_google_drive_file(service)
```
Comment (Billy4195) [inline arrange_schedule.py]: Typo `serach` on the above.

**Comment by icecat2012** [comment]
Done, please review when free.

File: arrange_schedule.py [hunk 12]
```diff
@@ -1035,7 +1035,126 @@ def check_event_exist_or_insert(event):
...
+        drive_file[num1]['time'] = 86400*7
+        for item in days_limit:
+            if item['id'] in drive_file[num1]['parents']:
+                drive_file[num1]['time'] = 86400 * int(item['name'][:-3])
+                break
+    return drive_file
+
+def check_drive_img_exist(db, data_type, file_name):
+    sql = "SELECT COUNT(*) FROM image_data WHERE img_is_expire=0 and img_is_delete=0 "
+    sql = sql + "and type_id={data_type} and img_file_name='{file_name}'".format(data_type=str(data_type), file_name=file_name)
```
Comment (Billy4195) [inline arrange_schedule.py]: The `data_type` does not need to cast to str(), because the format method will transform automatically

**Comment by Billy4195** [comment]
@chenyang14, @stvreumi could you help me to check this PR?

File: arrange_schedule.py [hunk 13]
```diff
@@ -1035,7 +1035,126 @@ def check_event_exist_or_insert(event):
...
-#crawler google drive image
+def find_drive_data_type():
+    db = mysql()
+    db.connect()
+    return_msg={}
+    return_msg['result'] = 'fail'
+    sql = "SELECT type_id FROM data_type WHERE type_name='google_drive_image'"
+    pure_result = db.query(sql)
+    try:
+        return_msg['data_type'] = int(pure_result[0][0])
```
Comment (chenyang14) [inline arrange_schedule.py]: The method getTypeId in DataTypeDao() has been used in this sql.

File: arrange_schedule.py [hunk 14]
```diff
@@ -1035,7 +1035,124 @@ def check_event_exist_or_insert(event):
...
+        fields="nextPageToken, files(id, name)").execute()
+    items = results.get('files', [])
+    return items
+
+def search_google_drive_file(service):
+    #set time
+    now_time = time.time()
+    start_time = time.strftime("%Y-%m-%dT%H:%M:%S+08:00", time.localtime(now_time-43200))
+    results = service.files().list(
+        q="modifiedTime > '" + start_time + "' and mimeType contains 'image/'",
```
Comment (stvreumi) [inline arrange_schedule.py]: Maybe line 1064 could change to
`"modifiedTime > '{start_time}' and mimeType contains 'image/'".format(start_time=start_time)`

**Pull request merged**